improver_tests/developer_tools/test_MOMetadataInterpreter.py

```diff
@@ -0,0 +1,409 @@
+    """Test interpretation of wind direction field with mean over realizations
+    cell method"""
+    interpreter.run(wind_direction_cube)
+    assert interpreter.diagnostic == "wind_from_direction"
+    assert interpreter.model == "MOGREPS-UK"
+    assert not interpreter.blended
+
+
+def test_weather_code(wxcode_cube, interpreter):
+    """Test interpretation of weather code field"""
+    interpreter.run(wxcode_cube)
+    assert interpreter.diagnostic == "weather_code"
+    assert interpreter.model == "UKV, MOGREPS-UK"
+    assert interpreter.blended
+
+
+## Test errors and warnings (input cubes in alphabetical order by fixture)
+
+
+def test_error_inconsistent_model_attributes(ensemble_cube, interpreter):
+    """Test error raised when the model ID and title attributes are inconsistent"""
+    ensemble_cube.attributes["mosg__model_configuration"] = "uk_det"
+    with pytest.raises(ValueError, match="Title.*is inconsistent with model ID"):
+        interpreter.run(ensemble_cube)
+
+
+def test_error_wrong_percentile_name_units(wind_gust_percentile_cube, interpreter):
+    """Test incorrect percentile coordinate name and units"""
+    wind_gust_percentile_cube.coord("percentile").units = "1"
+    wind_gust_percentile_cube.coord("percentile").rename("percentile_over_realization")
+    msg = (
+        ".*should have name percentile, has percentile_over_realization\n"
+        ".*should have units of %, has 1"
+    )
+    with pytest.raises(ValueError, match=msg):
+        interpreter.run(wind_gust_percentile_cube)
+
+
+def test_error_forbidden_attributes(wind_gust_percentile_cube, interpreter):
+    """Test error raised when a forbidden attribute is present"""
+    wind_gust_percentile_cube.attributes["mosg__forecast_run_duration"] = "wrong"
+    with pytest.raises(ValueError, match="Attributes.*include.*forbidden values"):
+        interpreter.run(wind_gust_percentile_cube)
+
+
+def test_error_missing_required_attribute(wind_gust_percentile_cube, interpreter):
+    """Test error raised when a mandatory attribute is missing"""
+    wind_gust_percentile_cube.attributes.pop("source")
+    with pytest.raises(ValueError, match="missing.*mandatory values"):
+        interpreter.run(wind_gust_percentile_cube)
+
+
+def test_error_missing_wind_gust_attribute(wind_gust_percentile_cube, interpreter):
+    """Test error when a wind gust percentile cube is missing a required attribute"""
+    wind_gust_percentile_cube.attributes.pop("wind_gust_diagnostic")
+    with pytest.raises(ValueError, match="missing .* required values"):
+        interpreter.run(wind_gust_percentile_cube)
+
+
+def test_warning_wind_gust_attribute_wrong_diagnostic(
+    wind_gust_percentile_cube, interpreter
+):
+    """Test a warning is raised if a known diagnostic-specific attribute is included
+    on an unexpected diagnostic"""
+    wind_gust_percentile_cube.rename("wind_speed")
+    interpreter.run(wind_gust_percentile_cube)
+    print(interpreter.warnings)
```

Comment:
There's a print here, but not in any of the other test functions. Suggest checking that this is as intended.

Reply:
Thanks - will remove.